src/Data/Text/Builder/Linear/Dec.hs

```diff
@@ -108,8 +108,12 @@ unsafePrependDec marr !off n
     digits = "00010203040506070809101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899"#
 {-# INLINEABLE unsafePrependDec #-}
 
+-- minBound = 1 `shiftL` (finiteBitSize a - 1) = -2^(finiteBitSize a - 1)
+-- Note that the last digit of 2^k forms a cycle for k≥ 1: 2,4,8,6
+-- Thus it is enough to compute rem (finiteBitSize a) 4, i.e. finiteBitSize a .&. 3
+-- This avoid the need for the more expansive rem n 10.
```

Comment:
```suggestion
-- This avoid the need for the more expensive rem n 10.
```

Reply:
Good catch. I ended reworking the complete comment.